src/test/util_string_tests.cpp

```diff
@@ -110,6 +126,24 @@ BOOST_AUTO_TEST_CASE(ConstevalFormatString_NumSpec)
     FailFmtWithError<2>("%1$*2$", err_term);
     FailFmtWithError<2>("%1$.*2$", err_term);
     FailFmtWithError<2>("%1$9.*2$", err_term);
+
+    // Non-parity between tinyformat and ConstevalFormatString.
+    // tinyformat throws but ConstevalFormatString does not.
+    BOOST_CHECK_EXCEPTION(tfm::format(ConstevalFormatString<1>{"%n"}, 0), tfm::format_error,
```

Comment:
these are basically TODOs for follow-up PRs, right?
Can we add `ConstevalFormatString` parity check for the [FUZZ tests](https://github.com/bitcoin/bitcoin/blob/master/src/test/fuzz/strprintf.cpp#L17) as well (guess that should reveal other cases where it's misaligned)

Reply:
I guess a version of `ConstevalFormatString/CheckNumFormatSpecifiers()` could be checked in a test together with `tfm::format`, but I'm not sure it would be a fuzz-test, as `tfm::format()` takes args specified at compile time.

It could conceivably be a pretty advanced fuzz-test that generated C++ code and compiled it as part of the test, not sure if we have that class of tests already or if this would be new, and if it would be accepted.

Reply:
Why not validate it during runtime via `CheckNumFormatSpecifiers`?

Reply:
`CheckNumFormatSpecifiers` is currently a template, but sure, could be changed to take the number of specifiers at runtime instead.

If we want to check *parity* in a fuzzing context we should probably check `tfm::format()` as well, with the same input?

Reply:
Didn't think to lock-down the number of args in that way and still get value from fuzzing. Turned out great!

Are you going to open a PR for this or what do you suggest as next step?